proxy/tun/dev/dev.linux.go

```diff
@@ -0,0 +1,150 @@
+// +build linux android
+
+package dev
+
+import (
+	"errors"
+	"net/url"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"github.com/google/netstack/tcpip/link/fdbased"
+	"github.com/google/netstack/tcpip/stack"
+)
+
+type tun struct {
+	name    string
+	fd      int
+	closeFd bool
+}
+
+func OpenTunDevice(deviceURL url.URL) (TunDevice, error) {
+	switch deviceURL.Scheme {
+	case "dev":
+		return openDeviceByName(deviceURL.Host)
+	case "fd":
+		fd, err := strconv.ParseInt(deviceURL.Host, 10, 32)
+		if err != nil {
+			return nil, err
+		}
+		return openDeviceByFd(int(fd))
+	}
+
+	return nil, errors.New("Unsupported device type " + deviceURL.Scheme)
+}
+
+func (t tun) Name() string {
+	return t.name
+}
+
+func (t tun) AsLinkEndpoint() (stack.LinkEndpoint, error) {
+	mtu, err := t.getInterfaceMtu()
+
+	if err != nil {
+		return nil, errors.New("Unable to get device mtu")
+	}
+
+	return fdbased.New(&fdbased.Options{
```

Comment:
Is this function safe to be called the second time? 
maybe `fdbased.New` can't be called twice?